src/test/deploy/functions/triggerRegionHelper.spec.ts

```diff
@@ -104,5 +104,27 @@ describe("TriggerRegionHelper", () => {
         region: "us",
       });
     });
+
+    it("should set trigger region from API then reject on invalid function region", async () => {
+      storageStub.resolves({ location: "US" });
+      const wantFn: backend.Endpoint = {
+        id: "wantFn",
+        entryPoint: "wantFn",
+        platform: "gcfv2",
+        eventTrigger: {
+          eventType: "google.cloud.storage.object.v1.finalized",
+          eventFilters: {
+            bucket: "my-bucket",
+          },
+          retry: false,
+        },
+        ...SPEC,
+        region: "europe-west4",
+      };
+
+      await expect(triggerRegionHelper.ensureTriggerRegions(backend.of(wantFn))).to.be.rejectedWith(
+        "A function in region europe-west4 cannot listen to a bucket in region us"
```

Comment:
My heart fluttered with excitement at the clarity we're adding to these complex rules.